tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11372,6 +11386,31 @@ bool hlsl::DiagnoseNodeStructArgument(Sema *self, TemplateArgumentLoc ArgLoc,
   }
 }
 
+// validates that for every function in the translation unit, if it
+// references a patch constant function, there exists a function declaration
+// that could serve as a candidate to that patch constant function.
+void ValidatePatchConstantFunctionsExist(clang::Sema *self) {
+  for (auto decl : self->getASTContext().getTranslationUnitDecl()->decls()) {
+    // TODO: improve condition so that only exported functions are checked,
+    // instead of all functions. Issue: #5857
+    if (FunctionDecl *FD = dyn_cast<FunctionDecl>(decl)) {
+      // If there is no patch constant function, then we don't need to validate
+      // anything.
+      if (const HLSLPatchConstantFuncAttr *Attr =
+              FD->getAttr<HLSLPatchConstantFuncAttr>()) {
```

Comment:
If I’m reading this correctly we call this function only on library shaders, then we iterate all functions looking for the patch constant attribute.

Do we somewhere else verify that the patch constant attribute is only present on entry functions marked as hull shaders?

Reply:
In short, yes. It's not accessible to godbolt because it's still stuck in staging-sm-6.8, but we have catches to determine both when this is inappropriately applied to other entry points and also when it is missing from the hull entry point. 

#5789 made sure that _something_ was specified as the patch constant function. This verifies that what was specified is a real function. I really don't know for sure if those checks should be one and the same. I can see an argument for separating the checks that you've declared _something_ and that the something you declared is actually valid. My gut reaction is to leave this as it is.

Reply:
@llvm-beanz, can you verify that this satisfies your concern?

Reply:
Yep. All good.